tests/stub/tx_run/test_tx_run.py

```diff
@@ -258,3 +258,28 @@ def test_failed_tx_run_allows_rollback(self):
 
     def test_failed_tx_run_allows_skipping_rollback(self):
         self._test_failed_tx_run(rollback=False)
+
+    @driver_feature(types.Feature.OPT_PULL_PIPELINING)
+    def test_should_terminate_tx_after_failed_run(self):
+        self._create_direct_driver()
+        self._server1.start(
+            path=self.script_path("tx_successful_and_failing_streams.script")
+        )
+        self._session = self._driver.session("r", fetch_size=2)
+        tx = self._session.begin_transaction()
+        res = tx.run("RETURN 1 AS n")
+        with self.assertRaises(types.DriverError) as exc:
+            tx.run("invalid")
+        self.assertEqual(exc.exception.code,
+                         "Neo.ClientError.Statement.SyntaxError")
+
+        # while already buffered records may be accessible, there must be no
+        # further PULLs and an exception must be raised
+        with self.assertRaises(types.DriverError):
+            for _i in range(0, 3):
+                res.next()
```

Comment:
Javascript start to consume the buffer data at this moment , so it tries to get another `PULL` and the script fails since it expects a `RESET`.

However, the result should be closed along with the failed transaction.